src/main/java/executor/command/CommandConvert.java

```diff
@@ -0,0 +1,194 @@
+ */
+public class CommandConvert extends Command {
+    private String from = " ";
+    private String to = " ";
+    private Double amount;
+    private Double exchangeRate = 0.00;
+
+    /**
+     * Constructor for the CommandConvert class.
+     * @param userInput this is the userInput from the CLI
+     */
+    public CommandConvert(String userInput) {
+        this.userInput = userInput;
+        this.commandType = CommandType.CONVERT;
+        this.amount = extractAmount(this.commandType, userInput);
+        this.from = getFromUserInput(userInput);
+        this.to = getToUserInput(userInput);
+    }
+
+
+    /**
+     * execute is the overall function which calls the converter and ui.
+     * @param wallet wallet is the overall wallet object
+     */
+    @Override
+    public void execute(Wallet wallet) {
+        Double convertedAmount = this.convertCurrency(this.getFrom(), this.getTo(), this.getAmount());
+        Ui.dukeSays(this.result(convertedAmount) + rateUsed());
+        Ui.printSeparator();
+    }
+
+
+    @Override
+    public void execute(TaskList taskList) {
+
+    }
+
+
+    /**
+     * extractAmount parses the user input from CLI to get the amount which the user wishes to convert.
+     * @param commandType this is the type of command
+     * @param userInput takes the user entered input from CLI
+     * @return the amount which user wishes to convert is returned
+     */
+    private Double extractAmount(CommandType commandType, String userInput) {
+        String amountStr = Parser.parseForPrimaryInput(commandType, userInput);
+        return Double.parseDouble(amountStr);
+    }
+
+    /**
+     * getFromUserInput parses user input for the flag "from".
+     * @param userInput this is the user entered input from CLI
+     * @return this function returns the 3 character unique string
+     */
+    private String getFromUserInput(String userInput) {
+        String fromStr = Parser.parseForFlag("from", userInput);
+        return fromStr;
+    }
+
+    /**
+     * getToUserInput parses the user input for the flag "to".
+     * @param userInput this is the user entered input from CLI
+     * @return this function return the 3 character unique string
+     */
+    private String getToUserInput(String userInput) {
+        String toStr = Parser.parseForFlag("to", userInput);
+        return toStr;
+    }
+
+    /**
+     * getTheJson is a function which gets the JSON object from the api.
+     * @param from this is the country from which we are converting currency
+     * @param to this is the country to which we are converting the currency
+     * @return this function returns a string version of the json object or else it will return null
+     */
+    private String getTheJson(String from, String to) {
+        try {
+            URL url = new URL("https://api.ratesapi.io/api/latest?base=" + from + "&symbols=" + to);
+            BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()));
```

Comment:
Consider refactoring _br_ to _reader_ for greater code clarity.

Reply:
refactored as requested.